src/csharp/Grpc.Core/ForwardedTypes.cs

```diff
@@ -0,0 +1,56 @@
+#region Copyright notice and license
+
+// Copyright 2019 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#endregion
+
+using System.Runtime.CompilerServices;
+using Grpc.Core;
+using Grpc.Core.Logging;
+using Grpc.Core.Utils;
+
+// API types that used to be in Grpc.Core package, but were moved to Grpc.Core.Api
+// https://docs.microsoft.com/en-us/dotnet/framework/app-domains/type-forwarding-in-the-common-language-runtime
+
+// TODO(jtattermusch): move types needed for implementing a client
+// TODO(jtattermusch): ServerServiceDefinition depends on IServerCallHandler (which depends on other stuff)
```

Comment:
Is it possible to change the codegen so we can optionally disable the generation of the method
```
    /// <summary>Creates service definition that can be registered with a server</summary>
    /// <param name="serviceImpl">An object implementing the server-side handling logic.</param>
    public static grpc::ServerServiceDefinition BindService(CounterBase serviceImpl)
    {
      return grpc::ServerServiceDefinition.CreateBuilder()
          .AddMethod(__Method_IncrementCount, serviceImpl.IncrementCount)
          .AddMethod(__Method_AccumulateCount, serviceImpl.AccumulateCount).Build();
    }
```
We would only pass in the flag for ASP.NET apps and that would mean we don't need the `ServerServiceDefinition` type to be moved to Grpc.Core.Api.

Reply:
It is possible to add an option, but I think it's not what we want.
- supporting more options in Grpc.Tools adds complexity and makes delivering "one size fits all" solution harder
- relying on codegen option will make it impossible to use both Grpc.Core server and Grpc.AspNetCore.Server in the same project (one requires the BindService() method and the other one forbids it).

I've done some experimenting and I think I'll be able to refactor ServerServiceDefinition so it can be moved to Grpc.Core.Api as well - so far I'm quite happy with my WIP and I think I'll have a PR for that soon.